### PR TITLE
Fix fmt.Stringer.String() called on nil pointer

### PR DESCRIPTION
**Pull request by wilhelmliao**
## Summary
got error when call bind() using parameter `(*fmt.Stringer)(nil)`.

```golang
var v *json.Number  // fmt.Stringer
query := "SELECT ?"
params := []interface{}{
v,
}
sql, err := bindPositional(time.UTC, query, params...)
fmt.Println(sql)
```
want
```
SELECT NULL
```
got
```
panic: value method encoding/json.Number.String called using nil *Number pointer
```

## Checklist
Delete items not relevant to your PR:
- [v ] Unit and integration tests covering the common scenarios were added

### CODE REVIEW

File: bind_test.go [hunk 1]
```diff
@@ -18,11 +18,13 @@
 package clickhouse
 
 import (
+	"encoding/json"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.openly.dev/pointy"
```
Comment (jkaflik) [inline bind_test.go:27]: let's avoid introducing additional dependency

Comment (wilhelmliao) [reply bind_test.go]: fixed

**Comment by jkaflik** [comment]
@wilhelmliao could you remove conflicts?

**Comment by wilhelmliao** [comment]
I fixed the conflicts, but the patched codes seem missing.

https://github.com/wilhelmliao/clickhouse-go/blob/main/bind.go#L308:313
```golang
	case fmt.Stringer:
		rv := reflect.ValueOf(v)
		if rv.Kind() == reflect.Pointer {
			if rv.IsNil() {
				return "NULL", nil
			}
		}
		return quote(v.String()), nil
```

**Comment by jkaflik** [comment]
@wilhelmliao yes, there was a PR improving bind - https://github.com/ClickHouse/clickhouse-go/pull/1092.

Thanks for your effort!

**Pull request closed without merging**